src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java

```diff
@@ -359,27 +321,20 @@ public FormValidation doTestConnection(
                 if (jenkinsToBitbucketCredentials == null) {
                     Guice.createInjector(new JenkinsToBitbucketCredentialsModule()).injectMembers(this);
                 }
+                BitbucketClientFactory client = clientFactoryProvider
+                        .getClient(
+                                config.getBaseUrl(),
+                                jenkinsToBitbucketCredentials.toBitbucketCredentials(config.getAdminCredentials()));
                 Optional<String> username =
-                        clientFactoryProvider
-                                .getClient(
-                                        config.getBaseUrl(),
-                                        jenkinsToBitbucketCredentials.toBitbucketCredentials(config.getAdminCredentials(), config.getGlobalCredentialsProvider(context)))
+                        client
                                 .getAuthenticatedUserClient()
                                 .getAuthenticatedUser();
                 if (!username.isPresent()) {
-                    return FormValidation.error("We can't connect to Bitbucket Server. Choose a different personal access token with project admin permissions");
+                    return FormValidation.error(
+                            "We can't connect to Bitbucket Server. Choose a different personal access token with project admin permissions");
                 }
-                BitbucketClientFactory client =
-                        clientFactoryProvider.getClient(
-                                config.getBaseUrl(),
-                                jenkinsToBitbucketCredentials.toBitbucketCredentials(credentials.orElse(null), config.getGlobalCredentialsProvider(context)));
 
                 AtlassianServerCapabilities capabilities = client.getCapabilityClient().getServerCapabilities();
-                if (credentials.filter(StringCredentials.class::isInstance).isPresent()) {
```

Comment:
Why is this removed?

Reply:
Well, we no longer have `credentials` in the server config. Is there any other way to get the credentials for this check?

Reply:
P.S. why was this permission check limited to `StringCredentials` only?

Reply:
This was checking if your token worked. `StringCredentials` are for bearer tokens, and that was how we used to check which of them failed, but you are probably right, it can go now, as we _know_ it is the PAT that's broken if we can't authenticate against the server.